backend/controllers/authentication.go

```diff
@@ -0,0 +1,97 @@
+package controllers
+
+import (
+	"backend/config"
+	"net/http"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+)
+
+// login body data definiton
+type Login struct {
+	Username string `form:"username" json:"username" binding:"required"`
+	Password string `form:"password" json:"password" binding:"required"`
+}
+
+// login controller supports two encoding: json and x-www-form-urlencoded
+func LoginController(c *gin.Context) {
+	var body Login
+	invalidFormatResponse := gin.H{"error": "invalid format"}
+	switch c.ContentType() {
+	case "application/json":
+		if err := c.ShouldBindBodyWithJSON(&body); err != nil {
+			c.JSON(http.StatusUnauthorized, invalidFormatResponse)
+			return
+		}
+
+	case "application/x-www-form-urlencoded":
+		if err := c.ShouldBind(&body); err != nil {
+			c.JSON(http.StatusUnauthorized, invalidFormatResponse)
+			return
+		}
+	}
+
+	// TODO: REMOVE DUMMY ASSESSMENT AND VALIDATE HASHED PASSWORD WITH DATABASE
+	// ok := util.CompareHashAndPassword(body.Password, <hashed_password>)
+
+	// dummy login assessment for now, to remove
+	if body.Username != "user" || body.Password != "password" {
```

Comment:
Check the previous comment about login and verify its structure with @cl4414.

Reply:
Fixed, thanks for the notice